### PR TITLE
Show UserAddress on the the LocationButton.

### PR DESCRIPTION
**Pull request by MohamedSaiko**
Requesting Code Review.

Simulator Screen Shot - iPhone 13 Pro - 2024-03-24 at 13 30 19

### CODE REVIEW

File: iStore/Home/HomeViewModel.swift [hunk 1]
```diff
@@ -19,24 +18,22 @@ final class HomeViewModel: ObservableObject {
...
-            case .failure(let error):
-                print(NetworkError.unknownError(error))
+                case .success(let data):
+                    DispatchQueue.main.async {
+                        self.products.append(contentsOf: data.products)
+                        self.isloading = false
+                    }
+                    
+                case .failure(let error):
+                    print(NetworkError.unknownError(error))
```
Comment (MohamedSaiko) [inline iStore/Home/HomeViewModel.swift:36]: no need for reviewing this.

File: iStore/Home/HomeViewModel.swift [hunk 2]
```diff
@@ -19,24 +18,22 @@ final class HomeViewModel: ObservableObject {
...
-            case .success(let data):
-                DispatchQueue.main.async {
-                    self.products.append(contentsOf: data.products)
-                    self.isloading = false
-                }
-                
-            case .failure(let error):
-                print(NetworkError.unknownError(error))
+                case .success(let data):
+                    DispatchQueue.main.async {
```
Comment (akabel92) [inline iStore/Home/HomeViewModel.swift]: Why are you using main queue here?

Comment (MohamedSaiko) [reply iStore/Home/HomeViewModel.swift]: Because It updates the UI.

UI updates should be on the main thread.

Comment (akabel92) [reply iStore/Home/HomeViewModel.swift]: But you are in the view model. Why do you want to update the UI from the view model?

Comment (MohamedSaiko) [reply iStore/Home/HomeViewModel.swift]: why wouldn't I ?
Isn't the viewModel meant for removing the logic from the view ?

Comment (akabel92) [reply iStore/Home/HomeViewModel.swift]: You should do all logic in view model as you said. When you want to refresh the view and use the main thread, you should make it in the view not view model.

**Review by akabel92 — APPROVED** [review]

**Pull request merged**